zcash_client_backend/src/fees/common.rs

```diff
@@ -43,47 +51,98 @@ where
         .map(|t_out| t_out.value())
         .sum::<Option<_>>()
         .ok_or_else(overflow)?;
-    let sapling_in = sapling_inputs
+    let sapling_in = sapling
+        .inputs()
+        .iter()
+        .map(sapling_fees::InputView::<NoteRefT>::value)
+        .sum::<Option<_>>()
+        .ok_or_else(overflow)?;
+    let sapling_out = sapling
+        .outputs()
         .iter()
-        .map(|s_in| s_in.value())
+        .map(sapling_fees::OutputView::value)
         .sum::<Option<_>>()
         .ok_or_else(overflow)?;
-    let sapling_out = sapling_outputs
+
+    #[cfg(feature = "orchard")]
+    let orchard_in = orchard
+        .inputs()
         .iter()
-        .map(|s_out| s_out.value())
+        .map(orchard_fees::InputView::<NoteRefT>::value)
         .sum::<Option<_>>()
         .ok_or_else(overflow)?;
+    #[cfg(not(feature = "orchard"))]
+    let orchard_in = NonNegativeAmount::ZERO;
+
+    #[cfg(feature = "orchard")]
+    let orchard_out = orchard
+        .outputs()
+        .iter()
+        .map(orchard_fees::OutputView::value)
+        .sum::<Option<_>>()
+        .ok_or_else(overflow)?;
+    #[cfg(not(feature = "orchard"))]
+    let orchard_out = NonNegativeAmount::ZERO;
+
+    // TODO: implement a less naive strategy for selecting the pool to which change will be sent.
+    #[cfg(feature = "orchard")]
+    let (change_pool, sapling_change, orchard_change) =
+        if orchard_in.is_positive() || orchard_out.is_positive() {
+            // Send change to Orchard if we're spending any Orchard inputs or creating any Orchard outputs
+            (ShieldedProtocol::Orchard, 0, 1)
+        } else if sapling_in.is_positive() || sapling_out.is_positive() {
+            // Otherwise, send change to Sapling if we're spending any Sapling inputs or creating any
+            // Sapling outputs, so that we avoid pool-crossing.
+            (ShieldedProtocol::Sapling, 1, 0)
+        } else {
+            // For all other transactions, send change to Sapling.
+            // FIXME: Change this to Orchard once Orchard outputs are enabled.
```

Comment:
Added #1101

Reply:
This change caused a clippy lint (identical blocks in both conditional cases), but I think that's fine for now if we're going to address 1101 soon.